### PR TITLE
refactor: simplify isIterable interface

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* next
  * **PR #104** <a href="https://app.graphite.dev/github/pr/zefiros-software/axioms/104" target="_blank">Graphite</a> 
    * **PR #105** <a href="https://app.graphite.dev/github/pr/zefiros-software/axioms/105" target="_blank">Graphite</a> 
      * **PR #106** <a href="https://app.graphite.dev/github/pr/zefiros-software/axioms/106" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/zefiros-software/axioms/106?utm_source=stack-comment).

File: src/guard/is-iterable/is-iterable.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type { Traversable } from '../../type'
 
-export function isIterable<T, R, O>(x: O | Traversable<T, R>): x is Traversable<T, R> {
+export function isIterable<T, R>(x: Traversable<T, R> | unknown): x is Traversable<T, R> {
```
Comment (MickVanDuijn) [inline src/guard/is-iterable/is-iterable.ts:3]: This breaks inference of `T` and `R`

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/zefiros-software/axioms/106?utm_source=stack-comment).


Job Reference: Uz89iC4BUIYbA80cXO6m

**Pull request merged**